lib/datasource.js

```diff
@@ -513,6 +513,9 @@ DataSource.prototype.setupDataAccess = function(modelClass, settings) {
       idProp.type = idType;
       modelClass.definition.rawProperties[idName].type = idType;
       modelClass.definition.properties[idName].type = idType;
+      if (['oracle', 'mysql'].indexOf(this.connector.name) !== -1) {
```

Comment:
Is there any reason that you are setting `forceId` just for `oracle` and `mysql`?